backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/snapshots/ColdMergeSnapshotSingleDiskCommand.java

```diff
@@ -65,6 +63,10 @@ public void handleFailure() {
 
     @Override
     public boolean performNextOperation(int completedChildCount) {
+        if (getParameters().getCommandStep() == null) {
```

Comment:
any reason not to call this block from executeCommand?

Reply:
Nope, just aligned it with `RemoveSnapshotSingleDiskLiveCommand`

Reply:
to be honest I'm not sure what this comment means[1], but afaik getVolumeInfo() is a synchronous call, and calling it from performNextOperation can actually block the callback thread which may block other commands from making progress



[1] I suspect it may have been blindly copied from the live counterpart which was the first usage of coco

Reply:
Hm okay.
Do we want maybe to move this to executeCommand in the live command as well?

Reply:
> Hm okay. Do we want maybe to move this to executeCommand in the live command as well?

let's consider that as a separate change (not part of this PR)

Reply:
@ahadas But since we have added getVolumeInfo to the live command as well, then maybe it should run in executeCommand there as well for the same reason?

Reply:
it makes sense but we need to reduce the risk if we want this to get in now
the 'live' flow is significantly less used - we remove snapshots for stateless vms, when exporting vms, etc. live merge is done only when removing a snapshot of running vm (which is done more now due to hybrid backup but still, I don't expect many backups to run in parallel but sequentially) and on live storage migration which is not done so frequently

Reply:
@ahadas 
Got it.
Done